cfme/tests/ansible/test_embedded_ansible_services.py

```diff
@@ -550,15 +569,29 @@ def test_ansible_group_id_in_payload(
     assert "group" in result_dict["manageiq"]
 
 
-@pytest.mark.provider([EC2Provider], scope="function")
-def test_embed_tower_exec_play_against_amazon(
+@pytest.mark.parametrize(
+    "credential", CREDENTIALS, ids=[cred[0] for cred in CREDENTIALS]
+)
+@pytest.mark.provider(
+    [RHEVMProvider, EC2Provider, VMwareProvider], selector=ONE_PER_TYPE
+)
+@pytest.mark.uncollectif(
```

Comment:
**Question** is this `uncollectif` necessary? Just curious, I would think that the parametrization would not try to mix credential and provider.

Reply:
@john-dupuy `uncollectif` is required as it is creating `3 x 3 test cases` but I wanted for e.g. EC2 credentials only EC2provider should be there not VMware.

Reply:
Alternative is to directly implement pytest_generate_tests, its a grey area on the complexity of these statements. Here I'd vote for doing it this way.